pyaedt/edb_core/hfss.py

```diff
@@ -706,3 +713,475 @@ def get_layout_bounding_box(self, layout=None, digit_resolution=6):
+            # Need to account for the coax port dimension
+            bb = loi.GetBBox()
+            ll = [bb.Item1.X.ToDouble(), bb.Item1.Y.ToDouble()]
+            ur = [bb.Item2.X.ToDouble(), bb.Item2.Y.ToDouble()]
+            # dim = 0.26 * max(abs(UR[0]-LL[0]), abs(UR[1]-LL[1]))  # 0.25 corresponds to the default 0.5
+            # Radial Extent Factor, so set slightly larger to avoid validation errors
+            dim = 0.30 * max(abs(ur[0] - ll[0]), abs(ur[1] - ll[1]))  # 0.25 corresponds to the default 0.5
+            terms_bbox.append(self._edb.Geometry.PolygonData(ll[0] - dim, ll[1] - dim, ur[0] + dim, ur[1] + dim))
+        return self._edb.Geometry.PolygonData.GetBBoxOfPolygons(convert_py_list_to_net_list(terms_bbox))
+
+    @pyaedt_function_handler()
+    def get_ports_number(self):
+        """Return the total number of excitation ports in a layout.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        int
+            The number of ports.
+
+        """
+        port_list = []
+        for term in self._active_layout.Terminals:
+            if str(term.GetBoundaryType()) == "PortBoundary":
+                if "ref" not in term.GetName():
+                    port_list.append(term)
+        return len(port_list)
+
+    @pyaedt_function_handler()
+    def layout_defeaturing(self, simulation_setup=None):
+        """Defeature layout with reducing the number of points for polygons based on surface deviation criteria.
+
+        Parameters
+        ----------
+        simulation_setup :
+            Edb_DATA.SimulationConfiguration object
+
+        Returns
+        -------
+        bool
+            True when succeeded, False when failed.
+
+        """
+        if not isinstance(simulation_setup, SimulationConfiguration):
+            self._logger.error("Layout defeaturing requires an EDB_Data.SimulationConfiguration object as argument.")
+            return False
+        self._logger.info("Starting Layout Defeaturing")
+        polygon_list = self._pedb.core_primitives.polygons
+        #polygon_with_voids = self._pedb.core_layout.get_poly_with_voids(polygon_list)
+        self._logger.info("Number of Polygons with Voids Found: {0}".format(str(polygon_with_voids.Count)))
+        for _poly in polygon_list:
+            voids_from_current_poly = _poly.Voids
+            new_poly_data = self._pedb.core_layout.defeature_polygon(setup_info=simulation_setup, poly=_poly)
+            _poly.SetPolygonData(new_poly_data)
+            if len(voids_from_current_poly) > 0:
+                for void in voids_from_current_poly:
+                    void_data = void.GetPolygonData()
+                    if void_data.Area() < float(simulation_setup.minimum_void_surface):
+                        void.Delete()
+                        self._logger.warning(
+                            "Defeaturing Polygon {0}: Deleting Void {1} area is lower than the minimum criteria".format(
```

Comment:
```suggestion
                            "Defeaturing polygon {0}: Deleting void {1} area because it is lower than the minimum criteria".format(
```